File: Python/01/src/guild_admin.py
import discord
from discord.ext import commands
from config import config
import requests
import os
from src.functions.main import get_prefix

class guild_admin(commands.Cog):
    def __init__(self, client):
        self.client = client

    @commands.command()
    @commands.has_permissions(administrator=True)
    async def kick(self, ctx, member: discord.Member=None, *, reason=None):
        if member is None:
            return await ctx.reply(embed=discord.Embed(
                title=f"โปรดระบุผู้ใช้ที่ต้องการเตะด้วยนะคะ",
                color=0x00ffff
            ).set_author(
                name="ไม่สามารถดำเนินการได้ค่ะ!",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))
        await member.kick(reason=reason)
        await ctx.reply(embed=discord.Embed(
            title=f"`{member}`\nได้ถูกเตะออกจากดิสเรียบร้อยค่ะ",
            description="เหตุผล : " + ("None" if reason is None else reason),
            color=0x00ffff
        ).set_author(
            name="ดำเนินการเรียบร้อยค่ะ!",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))

    @commands.command()
    @commands.has_permissions(administrator=True)
    async def ban(self, ctx, member: discord.Member=None, *, reason=None):
        if member is None:
            return await ctx.reply(embed=discord.Embed(
                title=f"โปรดระบุผู้ใช้ที่ต้องการแบนด้วยนะคะ",
                color=0x00ffff
            ).set_author(
                name="ไม่สามารถดำเนินการได้ค่ะ!",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))
        await member.ban(reason=reason)
        await ctx.reply(embed=discord.Embed(
            title=f"`{member}`\nได้ถูกแบนออกจากดิสเรียบร้อยค่ะ",
            description="เหตุผล : " + ("None" if reason is None else reason),
            color=0x00ffff
        ).set_author(
            name="ดำเนินการเรียบร้อยค่ะ!",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))

    @commands.command()
    @commands.has_permissions(administrator=True)
    async def unban(self, ctx, *, member=None):
        if member is None:
            return await ctx.reply(embed=discord.Embed(
                title=f"โปรดระบุผู้ใช้ที่ต้องการปลดแบนด้วยนะคะ",
                color=0x00ffff
            ).set_author(
                name="ไม่สามารถดำเนินการได้ค่ะ!",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))

        banned_users = await ctx.guild.bans()
        member_name, member_discriminator = member.split('#')
        for ban_entry in banned_users:
            user = ban_entry.user

            if (user.name, user.discriminator) == (member_name, member_discriminator):
                await ctx.guild.unban(user)
                return await ctx.reply(embed=discord.Embed(
                    title=f"`{member}`\nได้ถูกปลดแบนเรียบร้อยค่ะ",
                    color=0x00ffff
                ).set_author(
                    name="ดำเนินการเรียบร้อยค่ะ!",
                    icon_url=self.client.user.avatar_url,
                    url=config.author_url
                ))

        await ctx.reply(embed=discord.Embed(
            title=f"ไม่พบผู้ใช้นี้ในรายชื่อผู้ที่ถูกแบนค่ะ",
            color=0x00ffff
        ).set_author(
            name="ไม่สามารถดำเนินการได้ค่ะ!",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))

    @commands.command(aliases=["edit-server-icon"])
    @commands.has_permissions(administrator=True)
    async def editsvi(self, ctx, url=None):
        if url is None:
            return await ctx.reply(embed=discord.Embed(
                title=f"โปรดระบุ url ของรูปด้วยนะคะ",
                color=0x00ffff
            ).set_author(
                name="ไม่สามารถดำเนินการได้ค่ะ!",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))

        open(f"./data/temp/icon_server_{ctx.guild.id}.webp", "wb").write(requests.get(ctx.guild.icon_url).content)
        
        await ctx.guild.edit(icon=requests.get(url).content)
        await ctx.reply(embed=discord.Embed(
            title=f"เปลี่ยนปกดิสตาม url เรียบร้อยค่ะ",
            color=0x00ffff
        ).set_author(
            name="ดำเนินการเรียบร้อยค่ะ!",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ).set_image(url=url))

    @commands.command(aliases=["restore-server-icon"])
    @commands.has_permissions(administrator=True)
    async def restoresvi(self, ctx):
        if not os.path.isfile(f"./data/temp/icon_server_{ctx.guild.id}.webp"):
            return await ctx.reply(embed=discord.Embed(
                title=f"ไม่มีข้อมูลปกดิสก่อนหน้านี้ของดิสนี้ค่ะ",
                color=0x00ffff
            ).set_author(
                name="ไม่สามารถดำเนินการได้ค่ะ!",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))

        await ctx.guild.edit(icon=open(f"./data/temp/icon_server_{ctx.guild.id}.webp", "rb").read())

        await ctx.reply(
            file=discord.File(f"./data/temp/icon_server_{ctx.guild.id}.webp", filename=f"icon_server_{ctx.guild.id}.webp"),
            embed=discord.Embed(
                    title=f"เปลี่ยนปกดิสเป็นรูปเดิมเรียบร้อยค่ะ",
                    color=0x00ffff
                ).set_author(
                    name="ดำเนินการเรียบร้อยค่ะ!",
                    icon_url=self.client.user.avatar_url,
                    url=config.author_url
                ).set_image(
                    url=f"attachment://icon_server_{ctx.guild.id}.webp"
                )
            )

    @commands.command(aliases=["edit-server-name"])
    @commands.has_permissions(administrator=True)
    async def editsvn(self, ctx, *, name=None):
        if name is None:
            return await ctx.reply(embed=discord.Embed(
                title=f"โปรดระบุชื่อดิสใหม่ที่ต้องการจะตั้งด้วยนะคะ",
                color=0x00ffff
            ).set_author(
                name="ไม่สามารถดำเนินการได้ค่ะ!",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))
        await ctx.guild.edit(name=name)
        await ctx.reply(embed=discord.Embed(
            title=f"เปลี่ยนชื่อดิสเป็น {name} เรียบร้อยค่ะ",
            color=0x00ffff
        ).set_author(
            name="ดำเนินการเรียบร้อยค่ะ!",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))

    @commands.command(aliases=["change-nickname"])
    @commands.has_permissions(administrator=True)
    async def chnick(self, ctx, member: discord.Member=None, name=None):
        if member is None or name is None:
            return await ctx.reply(embed=discord.Embed(
                title=f"โปรดระบุข้อมูลให้ครบถ้วนด้วยนะคะ",
                description=f"เช่น `{get_prefix(self.client, ctx)[0]}chnick @nicenathapong คนหล่อเท่`",
                color=0x00ffff
            ).set_author(
                name="ไม่สามารถดำเนินการได้ค่ะ!",
                icon_url=self.client.user.avatar_url,
                url=config.author_url
            ))

        await member.edit(nick=name)
        await ctx.reply(embed=discord.Embed(
            title=f"เปลี่ยนชื่อเล่น {member} เป็น {name} เรียบร้อยค่ะ",
            color=0x00ffff
        ).set_author(
            name="ดำเนินการเรียบร้อยค่ะ!",
            icon_url=self.client.user.avatar_url,
            url=config.author_url
        ))

def setup(client):
    client.add_cog(guild_admin(client))

File: JS/01/src/commands/guild_admin.js
const { MessageEmbed , GuildMember } = require('discord.js')

module.exports = [
    {
        name: "kick",
        category: "guild_admin",
        information: "เตะผู้ใช้",
        async run(client, message, args) {
            try {
                const member = message.mentions.members.first()
                if (!member) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "โปรดระบุผู้ใช้ที่ต้องการเตะด้วยนะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.guild.members.kick(member, (args[1] ? args.slice(1, args.length - 1).join(" ") : "None"))
                message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ดำเนินการเรียบร้อยค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: `\`${member.user.username + member.user.discriminator}\`\nได้ถูกเตะออกจากดิสเรียบร้อยค่ะ`,
                        description: "เหตุผล : " + (args[1] ? args.slice(1, args.length - 1).join(" ") : "None"),
                        color: 0x00ffff
                    })
                ]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "ban",
        category: "guild_admin",
        information: "แบนผู้ใช้",
        async run(client, message, args) {
            try {
                const member = message.mentions.members.first()
                if (!member) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "โปรดระบุผู้ใช้ที่ต้องการแบนด้วยนะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.guild.members.ban(member, {days: 7, reason: (args[1] ? args.slice(1, args.length - 1).join(" ") : "None")})
                message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ดำเนินการเรียบร้อยค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: `\`${member.user.username + member.user.discriminator}\`\nได้ถูกแบนออกจากดิสเรียบร้อยค่ะ`,
                        description: "เหตุผล : " + (args[1] ? args.slice(1, args.length - 1).join(" ") : "None"),
                        color: 0x00ffff
                    })
                ]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "unban",
        category: "guild_admin",
        information: "ปลดแบนผู้ใช้",
        async run(client, message, args) {
            try {
                const user = await client.users.fetch(args[0])
                if (!user) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "ไม่พบผู้ใช้นี้ในระบบค่ะ",
                        color: 0x00ffff
                    })
                ]})
                message.guild.members.unban(user, (args[1] ? args.slice(1, args.length - 1).join(" ") : "None"))
                message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ดำเนินการเรียบร้อยค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: `\`${user.username + user.discriminator}\`\nได้ปลดแบนจากดิสเรียบร้อยค่ะ`,
                        description: "เหตุผล : " + (args[1] ? args.slice(1, args.length - 1).join(" ") : "None"),
                        color: 0x00ffff
                    })
                ]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "edit-server-icon",
        aliases: ["editsvi"],
        category: "guild_admin",
        information: "เปลี่ยนไอคอนเซิร์ฟเวอร์ของคุณ",
        async run(client, message, args) {
            try {
                if (!args[0]) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "โปรดระบุ URL ของรูปภาพที่ต้องการจะเปลี่ยนด้วยนะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.guild.setIcon(args.join(" "))
                message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ดำเนินการเรียบร้อยค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: `เปลี่ยนไอคอนเซิร์ฟเวอร์ตาม url เรียบร้อยค่ะ`,
                        image: {
                            url: args.join(" ")
                        },
                        color: 0x00ffff
                    })
                ]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "edit-server-name",
        category: "guild_admin",
        information: "เปลี่ยนไอคอนของเซิร์ฟเวอร์",
        async run(client, message, args) {
            try {
                if (!args[0]) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "โปรดระบุชื่อเซิร์ฟเวอร์ที่ต้องการจะตั้งด้วยนะคะ",
                        color: 0x00ffff
                    })
                ]})
                message.guild.setName(args.join(" "))
                message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ดำเนินการเรียบร้อยค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: `เปลี่ยนชื่อเซิร์ฟเวอร์เป็น "${args.join(" ")}" เรียบร้อยค่ะ`,
                        color: 0x00ffff
                    })
                ]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    },
    {
        name: "change-nickname",
        aliases: ["chnick"],
        category: "guild_admin",
        information: "เปลี่ยนชื่อเล่นผู้ใช้",
        async run(client, message, args) {
            try {
                const member = message.mentions.members.first()
                if (!member) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "โปรดระบุผู้ใช้ที่ต้องการตั้งชื่อเล่นด้วยนะคะ",
                        color: 0x00ffff
                    })
                ]})
                if (!args[1]) return message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ไม่สามารถดำเนินการได้ค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: "โปรดระบุชื่อเล่นที่ต้องการจะตั้งค่าด้วยนะคะ",
                        description: `เช่น \`${await client.function.main.get_prefix(client, message)}chnick @nicenathapong คนหล่อเท่\``,
                        color: 0x00ffff
                    })
                ]})
                member.setNickname(args.slice(1, args.length).join(" "))
                message.loading.edit({embeds:[
                    new MessageEmbed({
                        author: {
                            icon_url: client.user.avatarURL(),
                            name: "ดำเนินการเรียบร้อยค่ะ!",
                            url: client.config.embed_author_url
                        },
                        title: `เปลี่ยนชื่อเล่น \`${member.user.username}\` เป็น \`${args.slice(1, args.length).join(" ")}\` เรียบร้อยค่ะ`,
                        color: 0x00ffff
                    })
                ]})
            } catch (e) {
                client.function.main.error_log(e, client, message)
            }
        }
    }
]